RecoParticleFlow/PFClusterProducer/interface/PFHGCalRecHitCreator.h

```diff
@@ -105,7 +125,9 @@ template <typename DET,PFLayer::Layer Layer,unsigned subdet>
  protected:
   edm::EDGetTokenT<HGCRecHitCollection> recHitToken_;
   std::string geometryInstance_;
-
+ private:
+  hgcal::RecHitTools recHitTools_;
+  std::vector<const CaloCellGeometryHGCALAdapter*> caloCells_;
```

Comment:
@rovere - you should consider using smart pointers here

Reply:
There is no need to use a smart pointer that I'm in charge of deleting. Moreover, the lifetime of this plugin spans the full job.